### PR TITLE
bugfix: download images from imageListWithAuth.yaml

### PR DESCRIPTION
**Pull request by kakaZhou719**
### Describe how you did it
add build flag `--image-list-with-auth` to support download container images from imageListWithAuth.yaml
example:
` sealer build -t abc:v1 --image-list-with-auth imageListWithAuth.yaml`

if it set, sealer will read its content and download extra container images to rootfs.

### CODE REVIEW

**Review by jsparter — COMMENTED** [review]
I have a question, The user images is scattered in the file `imageList`, `imageListWithAuth`, `application`. If the same image appears in all three places, is it not remove duplications yet?

**Comment by kakaZhou719** [comment]
> 

if target image is already exist in building container, will skip to download it.

**Comment by jsparter** [comment]
> > 
> 
> if target image is already exist in building container, will skip to download it.

I saw that `build/buildimage/utils.go FormatImages` is doing so, but there is no other func do this. So, what component skips duplications? `DefaultImageSaver`?

File: cmd/sealer/cmd/image/build.go [hunk 1]
```diff
@@ -78,6 +78,9 @@ func NewBuildCmd() *cobra.Command {
 		},
 	}
 	buildCmd.Flags().StringVarP(&buildFlags.Kubefile, "file", "f", "Kubefile", "Kubefile filepath")
+	//todo we can support imageList Flag to download extra container image rather than copy it to rootfs
+	buildCmd.Flags().StringVar(&buildFlags.ImageList, "image-list", "filepath", "`pathname` of imageList filepath, if set, sealer will read its content and download extra container")
```
Comment (Starnop) [inline cmd/sealer/cmd/image/build.go:82]: what's the difference for `image-list` and `image-list-with-auth`?

And how about combine them?

Comment (kakaZhou719) [reply cmd/sealer/cmd/image/build.go:82]: 1. image-list: extra container images list which is no auth required need to be saved to clusterimage. 
2. image-list-with-auth: extra container images list which is auth required need to be saved to clusterimage,  see example: http://sealer.cool/docs/getting-started/build-cloudimage.html#more-build-examples

Comment (Starnop) [reply cmd/sealer/cmd/image/build.go:82]: what's the purpose for store the image list at present?

**Comment by Starnop** [comment]
TODO: combine imageList and imageWithAuth @kakaZhou719

**Review by Starnop — APPROVED** [review]

**Pull request merged**